File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
// 1차 비밀지도

function solution(n, arr1, arr2) {
  const answer = Array(n).fill("");

  // 이진수 변환
  const binArray1 = [];
  const binArray2 = [];
  for (let i = 0; i < n; i++) {
    binArray1.push(arr1[i].toString(2).padStart(n, "0"));
    binArray2.push(arr2[i].toString(2).padStart(n, "0"));
  }

  // 규칙에 맞게 해독하기
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (binArray1[i][j] === "1" || binArray2[i][j] === "1") {
        answer[i] += "#";
      } else {
        answer[i] += " ";
      }
    }
  }

  return answer;
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    table1 = []
    table2 = []
    
    # 지도1 만들기
    for num in arr1:
        num = bin(num)[2:]
        num = "0" * (n - len(num)) + num
        table1.append(num)
        
    # 지도2 만들기
    for num in arr2:
        num = bin(num)[2:]
        num = "0" * (n - len(num)) + num
        table2.append(num)
    
    for i in range(n):
        tmp = ""
        for j in range(n):
            # 둘 중 하나라도 벽이면 벽
            if table1[i][j] == "1" or table2[i][j] == "1":
                tmp += "#"
            elif table1[i][j] == "0" and table2[i][j] == "0":
                tmp += " "
        answer.append(tmp)
    return answer